hack/versions/cmd/new/templates/upgrade_test.template

```diff
@@ -0,0 +1,201 @@
+/*
+Copyright 2020 The Skaffold Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package %PREV_VERSION%
+
+import (
+	"testing"
+
+	next "github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/latest/v1"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/yaml"
+	"github.com/GoogleContainerTools/skaffold/testutil"
+)
+
+func TestUpgrade(t *testing.T) {
```

Comment:
This is the one problem: this upgrade template should use the `expected` from the previous version.  We could extract this out with a reasonably easy sed script, I think?
```
sed -n '/expected := `apiVersion:/,/^`/p

Reply:
I am using "template variables" and replacing the them with the appropriate values. I thought the expected should be the next version after the upgrade no?

The verify function should then used the expected version from the latest/v1 to validate the upgrade test. (maybe I am misunderstanding?)

Reply:
But because this is a fixed template, future changes need to be reflected in here.  For example, the before section includes the secret-as-a-single-item:
```
      secret:
        id: id
        src: /file.txt
```

But from now on, this should be an array, right?